libs/jsonforms-components/src/lib/Controls/Inputs/inputTextControl.spec.tsx

```diff
@@ -352,4 +354,42 @@ describe('Input Text Control tests', () => {
       expect(formatSin(input)).toBe(expected);
     });
   });
+
+  describe('Rootschema validations', () => {
+    const schemaProps: GoAInputTextProps & ControlProps = {
+      uischema: textBoxUiSchema,
+      schema: {},
+      rootSchema: {
+        if: {},
+        then: {
+          required: ['firstName'],
+        },
+      },
+      handleChange: (path, value) => {},
+      enabled: true,
+      label: 'First Name',
+      id: 'firstName',
+      config: {},
+      path: 'firstName',
+      errors: '',
+      data: 'My Name',
+      visible: true,
+      isValid: true,
+      required: false,
+      isVisited: false,
+      setIsVisited: () => {},
+    };
+
+    it('root schema must exist', () => {
+      expect(schemaProps).toBeTruthy();
+      expect(schemaProps?.rootSchema).toBeTruthy();
+    });
+
+    it('Then condition should exist if there is a If condition', () => {
+      const rootSchema = schemaProps?.rootSchema as JsonSchema7;
+      expect(schemaProps).toBeTruthy();
+      expect(rootSchema?.if).toBeTruthy();
+      expect(rootSchema?.then).toBeTruthy();
+    });
+  });
```

Comment:
This isn't quite complete.  There should be some test that check data against the schema, i.e. showing that if the parent condition is matched, then the child element must have a value.